services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -250,16 +260,132 @@ class MetersSpec extends BaseSpecification {
-        30478    | 32000 //should have been 32001, but use the maximum 32000
+        flowRate << [
+            //flowRate below min
+            ((CENTEC_MIN_BURST - 1) / getBurstCoef()).toBigDecimal().setScale(0, RoundingMode.CEILING).toLong(),
+            1000, //casual middle value
+            //flowRate above max
+            ((CENTEC_MAX_BURST + 1) / getBurstCoef()).toBigDecimal().setScale(0, RoundingMode.CEILING).toLong()
+        ]
+        sw = getCentecSwitches()[0]
+        expectedBurstSize = getExpectedBurst(sw.dpId, flowRate)
+    }
+
+    @Unroll
+    def "System allows to reset meter values to defaults without reinstalling rules for #data.description flow"() {
+        given: "A flow with custom meter rate and burst, that differ from defaults"
+        def src = data.srcSwitch
+        def dst = data.dstSwitch
+        assumeTrue("Desired switch combination is not available in current topology", src && dst)
+        def flow = flowHelper.randomFlow(src, dst)
+        flow.maximumBandwidth = 1000
+        flowHelper.addFlow(flow)
+        /*at this point meters are set for given flow. Now update flow bandwidth directly via DB, so that existing meter
+        rate and burst is no longer correspond to the flow bandwidth*/
+        def newBandwidth = 2000
+        database.updateFlowBandwidth(flow.id, newBandwidth)
+        //at this point existing meters do not correspond with the flow
+        //now save some original data for further comparison before resetting meters
+        Map<SwitchId, SwitchFlowEntries> originalRules = [src.dpId, dst.dpId].collectEntries {
+            [(it): northbound.getSwitchRules(it)]
+        }
+        Map<SwitchId, List<MeterEntry>> originalMeters = [src.dpId, dst.dpId].collectEntries {
+            [(it): northbound.getAllMeters(it).meterEntries]
+        }
+        def defaultMeters = { it.meterId < Constants.MAX_DEFAULT_METER_ID }
+        def flowMeters = { it.meterId >= Constants.MAX_DEFAULT_METER_ID }
+
+        when: "Ask system to reset meters for the flow"
+        def response = northbound.resetMeters(flow.id)
+
+        then: "Response contains correct info about new meter values"
+        [response.srcMeter, response.dstMeter].each { switchMeterEntries ->
+            switchMeterEntries.meterEntries.each { meterEntry ->
+                assert meterEntry.rate == newBandwidth
+                assert meterEntry.burstSize == getExpectedBurst(switchMeterEntries.switchId, newBandwidth)
+                //id of the meter is the same
+                assert originalMeters[switchMeterEntries.switchId]
+                        .findAll(flowMeters)*.meterId.contains(meterEntry.meterId)
+            }
+        }
+
+        and: "Non-default meter rate and burst are actually changed to expected values both on src and dst switch"
+        [src.dpId, dst.dpId].each { SwitchId swId ->
+            northbound.getAllMeters(swId).meterEntries.findAll(flowMeters)
+                .each { meterEntry ->
```

Comment:
Maybe it is worth using sameBeanAs() to compare the getAllMeters() and resetMeters() results to avoid code duplication. Just a suggestion though, the code is fine as it is.

Reply:
done